apps/analytics/analytics/processing/tile_count.py

```diff
@@ -0,0 +1,130 @@
+import time
+import pandas as pd
+
+from typing import Dict, List
+
+from analytics.mongo import RawDbClient, AnalyticsDbClient
+
+from models.analytics import TileCount
+
+pd.set_option('display.max_columns', None)
+pd.set_option('display.width', 800) 
+pd.set_option('display.max_rows', 150)  # display up to 200 rows
+
+def calculate_tile_count(
+    username: str,
+    raw_db_client: RawDbClient,
+) -> TileCount:
+    game_tick_data = raw_db_client.get_game_tick_collection().find(
+        {
+            "username": username,
+        }
+    )
+
+    ### build initial DF
+    gtdf = pd.DataFrame(game_tick_data)
+
+    gtdf['energy'] = gtdf['data'].apply(lambda row: row['energy'])
+    gtdf['sessionTickCount'] = gtdf['data'].apply(lambda row: row['sessionTickCount'])
+    gtdf['x'] = gtdf['data'].apply(lambda row: row['location']['x'] if 'location' in row and 'x' in row['location'] else None)
+    gtdf['y'] = gtdf['data'].apply(lambda row: row['location']['y'] if 'location' in row and 'y' in row['location'] else None)
+    gtdf['regionId'] = gtdf['data'].apply(lambda row: row['location']['regionId'] if 'location' in row and 'regionId' in row['location'] else None)
+    gtdf['tile'] = list(zip(gtdf['x'], gtdf['y']))
+
+
+    gtdf.dropna(subset=['regionId'], inplace=True)
+    gtdf = gtdf.drop(columns=['data'])
+
+    ### change in x coord
+    gtdf['dx'] = (gtdf['x'].abs() - gtdf['x'].shift(1).abs()).abs().fillna(0)
+    ### change in y coord
+    gtdf['dy'] = (gtdf['y'].abs() - gtdf['y'].shift(1).abs()).abs().fillna(0)
+    ### change in timestamp - should roughly reflect tick length
+    gtdf['dt'] = (gtdf['timestamp'] - gtdf['timestamp'].shift(1)).abs().fillna(0)
+    ### change in regionId - large changes indicate non-walking movement i.e new area door, or teleportation 
+    gtdf['dr'] = (gtdf['regionId'] - gtdf['regionId'].shift(1)).abs().fillna(0)
+
+    ### calculate tiles moved
+    def calc_tiles_moved(row):
+        if row['dx'] == 0 and row['dy'] == 0:
+            return 0
+        elif (((row['dx'] > 2) or (row['dy'] > 2)) and row['dr'] > 0):
+            return 0
+        # elif (((row['dx'] > 2) or (row['dy'] > 2)) and row['dr'] == 0):
+        #     return abs(row['dx']) + abs(row['dy'])
+        elif (row['dx'] != 0) or (row['dy'] != 0):
+            return max(abs(row['dx']), abs(row['dy']))
+        else:
+            return "Error"
+        
+    gtdf['tiles_moved'] = gtdf.apply(calc_tiles_moved, axis=1)
```

Comment:
So every time you use .apply with a python function/lambda the iteration over each row will happen in python, and it will be quite slow

Won't be an issue till we have a lot more user data - but might be worth considering using data frame operations

Reply:
Like I think this one could just be

```python
gtdf["traveled_tiles"] = gtdf[["dx", "dy"]].abs().max(axis=1)
```

Or similar (sorry on my phone so I probs got some variables wrong)

Reply:
What about this as a full replacement for the block?


`gtdf['tiles_moved'] = gtdf[['dx', 'dy']].max(axis=1)`
`gtdf.loc[((gtdf['dx'] > 2) | (gtdf['dy'] > 2)) & (gtdf['dr'] != 0), 'tiles_moved'] = 0`

New output:
![image](https://github.com/Runelite-Wrapped/runelite-wrapped/assets/134611815/021929de-b284-4045-b876-7d3490ceee92)

Old output (conditional block)
![image](https://github.com/Runelite-Wrapped/runelite-wrapped/assets/134611815/663b40ff-5f80-4d2f-9622-b4ffdc902084)

Reply:
sweet